grails-app/migrations/0.8.x/changelog-2021-01-26-1100-migrate-price-into-product-price.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9 http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+
+  <!-- migrate existing product_package.price into product_price -->
+  <changeSet author="awalkowiak" id="2601202111001-1">
+    <preConditions onFail="MARK_RAN">
+      <and>
+        <columnExists tableName="product_package" columnName="price"/>
+        <columnExists tableName="product_price" columnName="product_package_id"/>
+      </and>
+    </preConditions>
+    <sql>
+      INSERT INTO product_price (id, price_type_code, price, product_package_id)
+        SELECT uuid(), 'DEFAULT_PRICE', price, id FROM product_package
+        WHERE price IS NOT NULL AND price > 0;
+
+      UPDATE product_package
+        INNER JOIN product_price ON product_price.product_package_id = product_package.id
+        SET product_package.product_price_id = product_price.id;
+    </sql>
+  </changeSet>
+
+  <!-- remove redundant product_package.price column  -->
+  <changeSet author="awalkowiak" id="2601202111001-2">
+    <preConditions onFail="MARK_RAN">
+      <columnExists tableName="product_package" columnName="price"/>
+    </preConditions>
+    <dropColumn tableName="product_package" columnName="price"/>
+  </changeSet>
+
+  <!-- remove temporary product_price.product_package_id column  -->
+  <changeSet author="awalkowiak" id="2601202111001-3">
+    <preConditions onFail="MARK_RAN">
+      <columnExists tableName="product_price" columnName="product_package_id"/>
+    </preConditions>
+    <dropColumn tableName="product_price" columnName="product_package_id"/>
```

Comment:
In the future we might make associations between product prices and other tables like product package, product supplier, product, etc. Would we want to keep this column just so we don't have to add it later? Or will the data migration to recrate the column be relatively simply?

Reply:
ProductPakcage has product_price_id, so it would be just adding a column for product_package_id and probably doing something like:
`UPDATE product_price`
`INNER JOIN product_package ON product_package.product_price_id = product_price.id`
`SET product_price.product_package_id = product_package.id;`